### PR TITLE
CY-324-Adding Alert for Realtime Protection & Props for Windows DNS

### PR DESCRIPTION
**Pull request by babhalodia**

### CODE REVIEW

File: cyences_app_for_splunk/default/props.conf [hunk 1]
```diff
@@ -469,3 +469,12 @@ EVAL-src = coalesce(cyences_src_ip, src, srcip)
 EXTRACT-dest_ip_extract = dst=(?<cyences_dest_ip>[^\s]+)
 EVAL-dest_ip = coalesce(cyences_dest_ip, dest_ip)
 EVAL-dest = coalesce(cyences_dest_ip, dest, dest_ip)
+
+
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/props.conf:473]: I've worked on DNS stuff before for Chapman

```
# Linux Bind Server DNS logs (with ISC Bind Add-on installed)
[isc:bind:query]
# index=dns
EXTRACT-src_ip_port = client\s+.*\s+(?<src>\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})#(?<src_port>\d+)
EXTRACT-query_and_questiontype = query:\s+(?<query>[^\s]+)\s+IN\s+(?<questiontype>[^\s]+)\s
# Domain extraction for domain .*?(?<domain>[^./:]+\.(\S{2,3}|\S{2,3}.\S{2,3}))$ in query
EVAL-message_type = "QUERY"
EVAL-record_type = if(questiontype="A", "ANY", questiontype)
EVAL-reply_code = "NoError"

# Linux Bind Server DNS logs (with ISC Bind Add-on installed)
[isc:bind:queryerror]
# index=dns
EXTRACT-src_ip_port = client\s+.*\s+(?<src>\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})#(?<src_port>\d+)
EXTRACT-query_and_questiontype_and_reply_code = query\s+failed\s+\((?<reply_code>[^\)]+)\)\s+for\s+(?<query>.+)\/IN\/(?<questiontype>[^\s]+)\s+at\s+
EVAL-message_type = "QUERY"
EVAL-record_type = if(questiontype="A", "ANY", questiontype)
EVAL-reply_code = if(isnull(reply_code), "Error", reply_code)


# Fix issue with query field value in Windows DNS data
[MSAD:NT6:DNS]
# index=msad
EVAL-query = replace(replace(replace(query, "\(\d+\)$", ""), "^\(\d+\)", ""), "\(\d+\)", ".")
# Domain extraction for domain .*?(?<domain>[^./:]+\.(\S{2,3}|\S{2,3}.\S{2,3}))$ in query
FIELDALIAS-EVAL =  AS|ASNEW <new_field_name>)+
EVAL-message_type = case(direction="Rcv" AND isnull(operation), "QUERY", direction="Snd" AND isnull(operation), "Query to external server", direction="Rcv" AND operation="R", "Response from external server", direction="Snd" AND operation="R", "RESPONSE", 1==1, message_type)
# Explaination for above statement - message_type
# =================================
# Example logs
# ----
#  ---- HEADERS ----                                             direction src        operation opcode 
# Line 40553: 2/10/2021 1:45:26 PM 12C8 PACKET  000002487FB27DD0 UDP Rcv 10.47.3.234     d5d6   Q [0001   D   NOERROR] AAAA   (6)config(5)teams(9)microsoft(3)com(0)
# Line 40555: 2/10/2021 1:45:26 PM 12C8 PACKET  000002480023C170 UDP Snd 8.8.8.8         0260   Q [0001   D   NOERROR] AAAA   (6)config(5)teams(9)microsoft(3)com(0)
# Line 40561: 2/10/2021 1:45:26 PM 12C8 PACKET  00000248010909E0 UDP Rcv 8.8.8.8         0260 R Q [8081   DR  NOERROR] AAAA   (6)config(5)teams(9)microsoft(3)com(0)
# Line 40563: 2/10/2021 1:45:26 PM 12C8 PACKET  000002487FB27DD0 UDP Snd 10.47.3.234     d5d6 R Q [8081   DR  NOERROR] AAAA   (6)config(5)teams(9)microsoft(3)com(0)
# ----
# Understanding
# ----
# Rcv + Query (Empty) → Client IP (DNS Server receive the request from the client)
# Snd + Query (Empty) → External DNS Server IP (DNS Server is sending a request to external DNS server)
# Rcv + Response (R) → External DNS Server IP (DNS server received a response from external server)
# Snd + Response (R) → Client IP (DNS server sending a final response to the client)
EVAL-record_type = if(questiontype="A", "ANY", questiontype)
```

File: cyences_app_for_splunk/default/eventtypes.conf [hunk 2]
```diff
@@ -60,3 +60,9 @@ search = sourcetype="o365:management:activity" Workload=AzureActiveDirectory Ope
 search = sourcetype="gapps:report:login" events{}.name IN ("login_success", "login_failure")
 # Note - For other events like login_challenge with passed status google is also sending login_success events, 
 # so we don't have to assign authentication tag to those events
+
+#######################
+#### Windows DNS Logs
+######################
+[dns_dc_queries]
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/eventtypes.conf:67]: I've worked on DNS stuff before for Chapman

```
[dns_dc_queries]
search = index=msad sourcetype="MSAD:NT6:DNS"

[dns_isc_queries]
search = index=dns (sourcetype="isc:bind:query" OR sourcetype="isc:bind:queryerror")
```

for tags.conf
```
[eventtype=dns_dc_queries]
network = enabled
resolution = enabled
dns = enabled

[dns_isc_queries]
network = enabled
resolution = enabled
dns = enabled
```

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 3]
```diff
@@ -717,7 +717,32 @@ action.summary_index._name = cyences
 action.summary_index.category = Windows
 action.summary_index.alert_name = Windows - Windows Process Tampering Detected
 
-
+[Windows - Realtime Protection Failed/Disabled]
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:720]: Please update the existing savedsearch instead
`Windows Defender - Windows Defender RealTime Protection Disabled or Failed`

File: cyences_app_for_splunk/default/tags.conf [hunk 4]
```diff
@@ -92,3 +92,12 @@ authentication = enabled
 ##############
 [eventtype=login_success_and_failure_events]
 authentication = enabled
+
+
+#######################
+#### Windows DNS Logs
+######################
+[eventtype=dns_dc_queries]
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/tags.conf:100]: from my previous work
```
[eventtype=dns_dc_queries]
network = enabled
resolution = enabled
dns = enabled

[dns_isc_queries]
network = enabled
resolution = enabled
dns = enabled
```

**Review by VatsalJagani — CHANGES_REQUESTED** [review]

File: cyences_app_for_splunk/default/eventtypes.conf (hunk 2 above)
Comment (babhalodia) [reply cyences_app_for_splunk/default/eventtypes.conf:67]: Here I am intentionally not adding index. But I will add the other one.

File: cyences_app_for_splunk/default/savedsearches.conf (hunk 3 above)
Comment (babhalodia) [reply cyences_app_for_splunk/default/savedsearches.conf:720]: We already have alert with above name. But it is not checking whether RealTime Protection is again enabled or not.

Comment (babhalodia) [reply cyences_app_for_splunk/default/savedsearches.conf:720]: Suggest any new name.

**Review by VatsalJagani — APPROVED** [review]

**Pull request merged**